framework/src/utils/Conversion.C

```diff
@@ -212,20 +215,31 @@ namespace Moose
   {
     switch (t)
     {
-    case EXEC_INITIAL:        return "INITIAL";
-    case EXEC_LINEAR:         return "LINEAR";
-    case EXEC_NONLINEAR:      return "NONLINEAR";
-    case EXEC_TIMESTEP_END:   return "TIMESTEP_END";
-    case EXEC_TIMESTEP_BEGIN: return "TIMESTEP_BEGIN";
-    case EXEC_CUSTOM:         return "CUSTOM";
-    case EXEC_FINAL:          return "FINAL";
-    case EXEC_FORCED:         return "FORCED";
-    case EXEC_FAILED:         return "FAILED";
-    case EXEC_SUBDOMAIN:      return "SUBDOMAIN";
-    case EXEC_NONE:           return "NONE";
+      case EXEC_INITIAL:        return "INITIAL";
+      case EXEC_LINEAR:         return "LINEAR";
+      case EXEC_NONLINEAR:      return "NONLINEAR";
+      case EXEC_TIMESTEP_END:   return "TIMESTEP_END";
+      case EXEC_TIMESTEP_BEGIN: return "TIMESTEP_BEGIN";
+      case EXEC_CUSTOM:         return "CUSTOM";
+      case EXEC_FINAL:          return "FINAL";
+      case EXEC_FORCED:         return "FORCED";
+      case EXEC_FAILED:         return "FAILED";
+      case EXEC_SUBDOMAIN:      return "SUBDOMAIN";
+      case EXEC_NONE:           return "NONE";
     }
     return "";
   }
+
+  template<>
+  std::string stringify(const Real & t)
+  {
+    std::ostringstream os;
+    // the correct value would be std::numeric_limits<Real>::max_digits10 (C++11)
+    os << std::scientific
+       << std::setprecision(std::numeric_limits<Real>::digits10 + 1)
```

Comment:
Ok, I'm fuzz testing this (with millions of random bit patterns for the floats) and `+ 1` seems to do the right thing. It is the threshold between not retaining and always retaining full precision (for `double`)